File: 003-largest-prime-factor/solution.js
isPrime = function(integer) {
  for (var counter = 2; counter < Math.floor(Math.sqrt(integer)); counter++) {
    if (integer % counter == 0) {
      return false;
    }
  }
  return true;
}

factors = function(integer) {
  var facs = [];
  for (var counter = 2; counter < Math.floor(Math.sqrt(integer)); counter++) {
    if (integer % counter == 0) {
      facs.push(counter);
    }
  }
  return facs;
}

facs = factors(600851475143).filter(function(factor) {
  return isPrime(factor);
})

console.log(facs[facs.length - 1])

File: 003-largest-prime-factor/Solution.java
class Solution {

  public static void main(String[] args) {
    long number = 600851475143L;
    int largestPrimeFactor = 0;
    for (int counter = 2; counter < (long) Math.sqrt(number); counter++) {
      if (number % counter == 0 && isPrime(counter)) {
        largestPrimeFactor = counter;
      }
    }
    System.out.println(largestPrimeFactor);
  }

  private static boolean isPrime(int number) {
    for (int counter = 2; counter <= (int) Math.sqrt(number); counter++) {
      if (number % counter == 0) {
        return false;
      }
    }
    return true;
  }

}